### PR TITLE
Add SPM.

### PR DESCRIPTION
**Pull request by bigMOTOR**

### CODE REVIEW

**Comment by bigMOTOR** [comment]
@freak4pc Hi, Could you take a look, please. 
Have a nice day!

**Comment by freak4pc** [comment]
Hey, did you check this actually works? There are some issues with SPM and RxSwift in some scenarios, see here:

https://bugs.swift.org/browse/SR-12303

**Comment by freak4pc** [comment]
Also can you check what's wrong with CI ?

**Comment by bigMOTOR** [comment]
@freak4pc Hi! Thank you for quick response. Sure it works. I live with SPM in prod, that is why I need SPM on this package )) I'm already switched to it on my branch.
I'll check CI? But do you have any idea `use of undeclared type 'KFCrossPlatformImage'` what could cause it because KFCrossPlatformImage valid type.

UPD: It was just an old Xcode 10 issue. Anyway CI still have some issues with ABI. I'll take a look later. Thx!

**Comment by bigMOTOR** [comment]
@freak4pc Hi! I hope you are doing great. I fixed CI due to SPM way. Let's push it live, SPM is really a must nowadays. 🙌
Have a good day!

File: Package.swift [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+      targets: ["RxKingfisher"]),
+  ],
+  dependencies: [
+    // Dependencies declare other packages that this package depends on.
+    .package(url: "https://github.com/ReactiveX/RxSwift.git", .upToNextMajor(from: "5.1.1" )),
+    .package(url: "https://github.com/onevcat/Kingfisher.git", .upToNextMajor(from: "5.15.7" )),
+  ],
+  targets: [
+    // Targets are the basic building blocks of a package. A target can define a module or a test suite.
+    // Targets can depend on other targets in this package, and on products in packages which this package depends on.
```
Comment (freak4pc) [inline Package.swift]: remove all the comments from this file

File: Package.swift [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  ],
+  products: [
+    // Products define the executables and libraries produced by a package, and make them visible to other packages.
+    .library(
+      name: "RxKingfisher",
+      targets: ["RxKingfisher"]),
+  ],
+  dependencies: [
+    // Dependencies declare other packages that this package depends on.
+    .package(url: "https://github.com/ReactiveX/RxSwift.git", .upToNextMajor(from: "5.1.1" )),
```
Comment (freak4pc) [inline Package.swift]: this should be 5.0.0 and up

File: Package.swift [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    .package(url: "https://github.com/onevcat/Kingfisher.git", .upToNextMajor(from: "5.15.7" )),
```
Comment (freak4pc) [inline Package.swift]: this should be 5.0.0 and up as well I believe unless there's a reason

File: Sources/RxKingfisher/Kingfisher+Rx.swift [hunk 4]
```diff
@@ -10,29 +10,29 @@ import RxCocoa
 import RxSwift
 import Kingfisher
 
-extension Reactive where Base == KingfisherWrapper<ImageView> {
+extension Reactive where Base == KingfisherWrapper<KFCrossPlatformImageView> {
```
Comment (freak4pc) [inline Sources/RxKingfisher/Kingfisher+Rx.swift]: if you have code changes to make you should do these in a separate PR, not this one - just to make sure we're reviewing things seperately. Thanks!

File: Sources/RxKingfisher/Kingfisher+Rx.swift [hunk 5]
```diff
@@ -10,29 +10,29 @@ import RxCocoa
...
-                         options: KingfisherOptionsInfo? = nil) -> Single<Image> {
+                         options: KingfisherOptionsInfo? = nil) -> Single<KFCrossPlatformImage> {
         return Single.create { [base] single in
             let task = base.setImage(with: source,
                                      placeholder: placeholder,
-                                     options: options) { result in
+                                     options: options, completionHandler: { result in
                 switch result {
                 case .success(let value):
                     single(.success(value.image))
```
Comment (freak4pc) [inline Sources/RxKingfisher/Kingfisher+Rx.swift]: revert

**Review by freak4pc — CHANGES_REQUESTED** [review]

**Comment by bigMOTOR** [comment]
Hi @freak4pc 
Thank you for your comments. All issues are resolved now.

**Review by freak4pc — APPROVED** [review]

**Pull request merged**